flocker/node/functional/test_docker.py

```diff
@@ -641,6 +644,28 @@ def test_restart_policy_on_failure_maximum_count(self):
         d.addCallback(self.assertIn, ("5", "6"))
         return d
 
+    def test_command_line(self):
+        """
+        A container with custom command line is run with those arguments.
+        """
+        directory = FilePath(self.mktemp())
+        directory.makedirs()
+        d = self.start_container(
+            random_name(), image_name=u"busybox",
+            volumes=[Volume(node_path=directory,
+                            container_path=FilePath(b"/data"))],
+            command_line=[u"sh", u"-c", u"echo 'hello' > /data/output"])
+
+        def started(_):
+            expected = directory.child(b"output")
+            for i in range(100):
+                if expected.exists():
+                    self.assertEqual(expected.getContent(), b"hello\n")
+                    return
```

Comment:
Doing a `loop_until(container is not running)` and then checking for the file might be more robust.  Not that this is related to how I saw this test fail on buildbot (with a timeout error, not a "Files never created" failure as happens below).